### ISSUE TITLE
Implementations // CS:GO & More

### ISSUE THREAD

**Issue by zomgasd** [cookkkie/mee6#18]
A script where you can implement "game state integration" where the bot can update a player joining a CS:GO server would be useful for example: There are now 4 players in the <location> server! [multiple server support] <- would be awesome.

An implementation of Twitch emoticons would be awesome, a random meme generators. Urban dictionary lookup would be amazing.

**Comment by jaw3l** [comment]
a command which brings SMITE builds, items, gods. this would be great

**Comment by sammyvsparks** [comment]
Gaming plugins are planned for some stuff, i'll add these to the suggestion list
+ CS:GO Server Status (If API is Available)
+ Urban Lookup
+ Twitch Emotes

Also a minecraft info plugin is planned

**Comment by zomgasd** [comment]
Can you give an estimate on how long for those update releases? @sammyvsparks

**Comment by SilBoydens** [comment]
no ETA on how long, will depend on how fast and how much Cookkkie can work on it. @zomgasd 

~ Support ; BoydensSil

**Comment by nstafie** [comment]
@zomgasd 

Game state integration is client-side and requires a CSGO client to be connected to a server, and it's the client that will be broadcasting the events, so that can't really be done.

However, what can be done is querying of 
1) players (how many players, their names, their scores and long they've been connected)
2) general server information (server name, map, number of players, number of bots, what OS the server is on, server visibility - public/private, VAC status - on/off and the version of the server)

With this information you can do things like channel topics updating with the server status, occasional chat messages with server info, and so on.

*However*, the 3 libraries that currently exist to do this, are either ancient, have no complete python3 support or don't use asyncio, meaning Cookkkie would have to write it from scratch.

tl;dr: live player join updates aren't possible, and what is possible takes a fair amount of effort. don't expect it too soon.

**Comment by zomgasd** [comment]
Thanks for the clarification @nstafie
